packages/component-library/src/components/plmg-text-input/plmg-text-input.tsx

```diff
@@ -0,0 +1,256 @@
+   *
+   * Allowed values:
+   * - Any string
+   *
+   * A unique label name for each element in a form is required for accessibility
+   */
+  @Prop() label: string;
+  @Watch('label')
+  validateLabelMessage(newValue: string) {
+    if (typeof newValue !== 'string' || !this.label)
+      throw new Error('label text is required. label text must be string');
+  }
+  /**
+   * Define if an input is required.
+   *
+   * Allowed values:
+   * - true
+   * - false
+   *
+   * Default: false
+   */
+  @Prop() required: boolean = false;
+  @Watch('required')
+  validateRequired(newValue: boolean) {
+    if (typeof newValue !== 'boolean')
+      throw new Error('required: must be boolean');
+  }
+  /**
+   *
+   * Define text input's size
+   *
+   * Allowed values:
+   *   - medium
+   *   - large
+   *
+   * Default: medium
+   */
+  @Prop() size: PlmgTextInputSize = 'medium';
+  @Watch('size')
+  validateSize(newValue: string) {
+    if (typeof newValue !== 'string' || newValue === '')
+      throw new Error('size: required');
+    if (!isPlmgTextInputSize(newValue))
+      throw new Error('size: must be a valid value');
+  }
+  /**
+   * Define tip text
+   *
+   * Allowed value: any string
+   *
+   * Displayed when tip is true
+   */
+  @Prop() tipText: string;
+  @Watch('tipText')
+  validateTipText(newValue: string) {
+    if (newValue && typeof newValue !== 'string')
+      throw new Error('tip text must be a string');
+  }
+  /**
+   * Define if tip text is displayed.
+   *
+   * Allowed values:
+   * - true
+   * - false
+   *
+   * Default: false
+   */
+  @Prop() tip: boolean = false;
+  @Watch('tip')
+  validateTipTextShow(newValue: boolean) {
+    if (typeof newValue !== 'boolean') throw new Error('tip must be boolean');
+  }
+  /**
+   * 3. State() variables
+   * Inlined decorator, alphabetical order.
+   *
+   * import { State } from '@stencil/core';
+   */
+  // @State() isValidated: boolean;
+
+  /**
+   * Reference to host HTML element.
+   */
+  @State() value: number | string;
+  @State() isError: boolean = false;
+
+  private handleInputChange(ev) {
+    this.value = ev.target.value;
+    this.valueUpdated.emit({ value: this.value });
+  }
+
+  @Event() valueUpdated: EventEmitter;
+  /**
+   * Life Cycle Methods & Event Listeners
+   */
+  connectedCallback() {
+    if (this.filled) {
+      this.value = this.default;
+    }
+  }
+
+  render() {
+    const inputClasses = {
+      [this.size]: true,
+      error: this.error,
+    };
+
+    const tipClasses = {
+      [this.size]: true,
+      tip: true,
+    };
+
+    const labelClasses = {
+      [this.size]: true,
+      ['label-visible']: this.LabelVisible,
+    };
+
+    return (
+      <div class={'plmg-text-input-wrapper'}>
+        <label
+          class={labelClasses}
+          htmlFor={this.labelToId()}
+          aria-label={this.label}
+        >
+          {this.showText()}
+          {this.showRequiredAsterix() && <span class={'required'}>*</span>}
```

Comment:
I suggest to scope these elements together to reduce complexity on showing the required indicator. We can first check for label visibility then nested we check for required visibility. There is no need to check for a presence of label text.  We should check with designers that there is no use case where we show the required indicator without a label.